jupyter_client/kernelspec.py

```diff
@@ -138,17 +138,10 @@ def find_kernel_specs(self):
         return d
         # TODO: Caching?
 
-    def get_kernel_spec(self, kernel_name):
-        """Returns a :class:`KernelSpec` instance for the given kernel_name.
-
-        Raises :exc:`NoSuchKernel` if the given kernel name is not found.
+    def get_kernel_spec_helper(self, kernel_name, resource_dir):
+        """ Returns a :class:`KernelSpec` instance for a given kernel_name
+        and resource_dir.
         """
-        d = self.find_kernel_specs()
-        try:
-            resource_dir = d[kernel_name.lower()]
```

Comment:
Doesn't this make it `get_kernel_spec_helper(self, kernel_name)` as it makes the passed in `ressource_dir` argument obsolete?